src/main/kotlin/snyk/oss/annotator/OSSBaseAnnotator.kt

```diff
@@ -16,15 +18,23 @@ import kotlin.math.max
 abstract class OSSBaseAnnotator : ExternalAnnotator<PsiFile, Unit>() {
     // overrides needed for the Annotator to invoke apply(). We don't do anything here
     override fun collectInformation(file: PsiFile): PsiFile = file
-    override fun doAnnotate(collectedInfo: PsiFile?): Unit = Unit
+    override fun doAnnotate(psiFile: PsiFile?) {
+        val filePath = psiFile?.virtualFile?.path ?: return
+
+        if (AnnotatorHelper.isFileSupported(filePath)) {
+            val document = psiFile.viewProvider.document ?: return
+            ApplicationManager.getApplication().invokeAndWait {
+                FileDocumentManager.getInstance().saveDocument(document)
+            }
+        }
+    }
 
     override fun apply(psiFile: PsiFile, annotationResult: Unit, holder: AnnotationHolder) {
-        val issues = getIssuesForFile(psiFile)
-        val remediation = issues?.remediation
+        val issues = getIssuesForFile(psiFile) ?: return
+        val remediation = issues.remediation
 
-        issues?.vulnerabilities
-            ?.distinctBy { it.id }
-            ?.forEach { vulnerability ->
+        issues.vulnerabilities
+            .forEach { vulnerability ->
```

Comment:
Removed the `distinctBy` to annotate all places where a vulnerability is caused.

Reply:
seems does not fix the issue:
![image](https://user-images.githubusercontent.com/14268320/153603075-1b618273-3769-4ac5-9c08-a35371f26efc.png)

Reply:
I think the current version fixes it. 
![image](https://user-images.githubusercontent.com/20150761/153637112-f150f158-21c4-47b4-8549-ec73a6e04c21.png)

Reply:
![image](https://user-images.githubusercontent.com/20150761/153637267-c3395988-ef4f-4756-a010-6f276eafdce3.png)
![image](https://user-images.githubusercontent.com/20150761/153637291-e1d1a74f-e0c7-4171-ad2a-307566441252.png)